Rel Notes for Storage 4.19 (#93118)
openshift/openshift-docs: release_notes/ocp-4-19-release-notes.adoc

[id="ocp-release-notes-storage_{context}"]
=== Storage

[id="ocp-release-notes-storage-azure-file-cross-sub-support_{context}"]

> ==== Azure File cross-subscription support is generally available

Reviewer: LGTM

> Cross-subscription support allows you to have an {product-title} cluster in one Azure subscription and mount your Azure file share in another Azure subscription using the Azure File Container Storage Interface (CSI) driver.

Reviewer 2: within the same tenant, OCP and the associated storage can be in different subcriptions but in the same Azure tenant

Author: Fixed.

This feature is generally available in {product-title} 4.19.

// For more information, see ???

[id="ocp-release-notes-storage-vol-attributes_{context}"]
==== Volume Attributes Classes (Technology Preview)
Volume Attributes Classes provide a way for administrators to describe "classes" of storage they offer. Different classes might correspond to different quality-of-service levels.

Volume Attributes Classes in {product-title} 4.19 is available only with AWS Elastic Block Storage (EBS) and Google Cloud Platform (GCP) persistent disk (PD) Container Storage Interface (CSI).

> You can apply a Volume Attributes Class to a persistent volume claim (PVC). If a new Volume Attributes Class becomes available in the cluster, a user can update the PVC with the new Volume Attributes Class if needed.

Reviewer 3: ```suggestion
You can apply a Volume Attributes Class to a persistent volume claim (PVC). If a new Volume Attributes Class becomes available in the cluster, you can update the PVC with the new Volume Attributes Class if needed.
```

> Volume Attribute Classes have parameters that describe volumes belonging to them. Different parameters can be accepted depending on the provisioner or the resizer. If a parameter is omitted, the default is used at volume provisioning. If a user applies the PVC with a different Volume Attributes Class with omitted parameters, the default value of the parameters might be used depending on the CSI driver implementation. For more information, see the related CSI driver documentation.

Reviewer 3: ```suggestion
Volume Attributes Classes have parameters that describe volumes belonging to them. Different parameters can be accepted depending on the provisioner or the resizer. If a parameter is omitted, the default is used at volume provisioning. If you apply the PVC with a different Volume Attributes Class with omitted parameters, the default value of the parameters might be used depending on the CSI driver implementation. For more information, see the related CSI driver documentation.
```
for consistency? referred to as plural "attributes" elsewhere

Volume Attributes Classes is available in {product-title} 4.19 with Technology Preview status.

//For more information, see ???.

[id="ocp-release-notes-storage-cli-cmd-pvc-usage_{context}"]
==== New CLI command to show PVC usage (Technology Preview)

> {product-title} 4.19 introduces a new command to view persistent volume claim usage. This feature has Technology Preview status.

Reviewer 3: ```suggestion
{product-title} 4.19 introduces a new command to view PVC usage. This feature has Technology Preview status.
```

// For more information, see ???.

[id="ocp-release-notes-storage-cli-cmd-resize-recovery_{context}"]
==== CSI volume resizing recovery is generally available

> Previously, you might expand a persistent volume claim (PVC) to a size that is not supported by the underlying storage provider. In which case, the expansion controller typically tries forever to expand the volume and keeps failing. 

Reviewer 3: ```suggestion
Previously, you might expand a PVC to a size that is not supported by the underlying storage provider. In this case, the expansion controller typically tries forever to expand the volume and keeps failing. 
```

This new feature allows you to recover and provide another resize value for the PVC. Resizing recovery is supported as generally available in {product-title} 4.19.

> For more information about resizing volumes, see xref:../storage/expanding-persistent-volumes.adoc[Expanding persistent volumes].

Reviewer 3: needs [anchor ID ](https://github.com/openshift/openshift-docs/blob/main/contributing_to_docs/doc_guidelines.adoc#links-hyperlinks-and-cross-references)

For more information about recovering when resizing volumes, see xref:../storage/expanding-persistent-volumes.adoc#expanding-recovering-from-failure_expanding-persistent-volumes[Recovering from failure when expanding volumes].

> [id="ocp-release-notes-storage-resize-migrated-vsphere-in-tree-vols_{context}"]

Reviewer: LGTM

==== Support for resizing vSphere in-tree migrated volumes is generally available

> Previously, VMware vSphere persistent volumes that were migrated from in-tree to Container Storage Interface (CSI) could not be resized. With {product-title} 4.19, resizing migrated volumes is supported. This feature is generally available.

Reviewer 3: ```suggestion
Previously, VMware vSphere persistent volumes that were migrated from in-tree to CSI could not be resized. With {product-title} 4.19, resizing migrated volumes is supported. This feature is generally available.
```

> For more information about resizing volumes, see xref:../storage/expanding-persistent-volumes.adoc[Expanding persistent volumes].

Reviewer 3: needs anchor ID

[id="ocp-release-notes-storage-disable-vsphere_{context}"]

> ==== Disabling and enabling storage on vSphere is generally available

Reviewer: LGTM

> Cluster administrators might want to disable the VMWare vSphere Container Storage Interface (CSI) Driver as a Day 2 operation, so the vSphere CSI Driver does not interface with your vSphere setup. 

Reviewer 3: ```suggestion
Cluster administrators might want to disable the VMWare vSphere CSI Driver as a Day 2 operation, so the vSphere CSI Driver does not interface with your vSphere setup. 
```

[id="ocp-release-notes-storage-increase-max-vols-per-node-vsphere"]
==== Increasing the maximum number of nodes per volume for vSphere (Technology Preview)
For VMware vSphere version 7, {product-title} restricts the maximum number of volumes per node to 59.

However, with {product-title} 4.19 for vSphere version 8 or later, you can increase the allowable number of volumes per node to a maximum of 255. Otherwise, the default value remains at 59.

This feature has Technology Preview status.

For more information, see xref:../storage/container_storage_interface/persistent-storage-csi-vsphere.adoc#persistent-storage-csi-vsphere-increase-max-vols-per-node-overview_persistent-storage-csi-vsphere[Increasing maximum volumes per node for vSphere].

[id="ocp-release-notes-storage-vsphere-migrating-cns-vols-between-datastores_{context}"]
==== Migrating CNS volumes between datastores for vSphere is generally available
If you are running out of space in your current datastore, or want to move to a more performant datastore, you can migrate VMware vSphere Cloud Native Storage (CNS) volumes between datastores. This applies to both attached and detached volumes.

> This features was introduced as a Development Preview in {product-title} 4.17, but is now generally available in 4.19.

Reviewer 3: ```suggestion
This feature was introduced as a Development Preview in {product-title} 4.17, but is now generally available in 4.19.
```

[id="ocp-release-notes-storage_{context}"]
=== Storage

[id="ocp-release-notes-storage-azure-file-cross-sub-support_{context}"]
==== Azure File cross-subscription support is generally available
Cross-subscription support allows you to have an {product-title} cluster in one Azure subscription and mount your Azure file share in another Azure subscription using the Azure File Container Storage Interface (CSI) driver. The subscriptions must be in the same tenant.

This feature is generally available in {product-title} 4.19.

// For more information, see ???

[id="ocp-release-notes-storage-vol-attributes_{context}"]

> ==== Volume Attributes Classes (Technology Preview)

Reviewer 4: LGTM

Volume Attributes Classes in {product-title} 4.19 is available only with AWS Elastic Block Storage (EBS) and Google Cloud Platform (GCP) persistent disk (PD) Container Storage Interface (CSI).

You can apply a Volume Attributes Classes to a persistent volume claim (PVC). If a new Volume Attributes Class becomes available in the cluster, you can update the PVC with the new Volume Attributes Classes if needed.

Volume Attributes Classes have parameters that describe volumes belonging to them. If a parameter is omitted, the default is used at volume provisioning. If a user applies the PVC with a different Volume Attributes Class with omitted parameters, the default value of the parameters might be used depending on the CSI driver implementation. For more information, see the related CSI driver documentation.

Volume Attributes Classes is available in {product-title} 4.19 with Technology Preview status.

//For more information, see ???.

[id="ocp-release-notes-storage-cli-cmd-pvc-usage_{context}"]
==== New CLI command to show PVC usage (Technology Preview)

> {product-title} 4.19 introduces a new command to view persistent volume claim usage. This feature has Technology Preview status.

Reviewer 5: LGTM

Previously, VMware vSphere persistent volumes that were migrated from in-tree to Container Storage Interface (CSI) could not be resized. With {product-title} 4.19, resizing migrated volumes is supported. This feature is generally available.

For more information about resizing volumes, see xref:../storage/expanding-persistent-volumes.adoc[Expanding persistent volumes].

[id="ocp-release-notes-storage-disable-vsphere_{context}"]
==== Disabling and enabling storage on vSphere is generally available
Cluster administrators might want to disable the VMWare vSphere Container Storage Interface (CSI) Driver as a Day 2 operation, so the vSphere CSI Driver does not interface with your vSphere setup. 

This features was introduced in {product-title} 4.17 with Technology Preview status. This feature is now supported as generally available in {product-title} 4.19.

For more information, see xref:../storage/container_storage_interface/persistent-storage-csi-vsphere.adoc#persistent-storage-csi-vsphere-disable-storage-overview_persistent-storage-csi-vsphere[Disabling and enabling storage on vSphere].

[id="ocp-release-notes-storage-increase-max-vols-per-node-vsphere"]

> ==== Increasing the maximum number of nodes per volume for vSphere (Technology Preview)

Reviewer 4: ```suggestion
==== Increasing the maximum number of volumes per node for vSphere (Technology Preview)
```

Author: Fixed.

For VMware vSphere version 7, {product-title} restricts the maximum number of volumes per node to 59.

However, with {product-title} 4.19 for vSphere version 8 or later, you can increase the allowable number of volumes per node to a maximum of 255. Otherwise, the default value remains at 59.

This feature has Technology Preview status.

For more information, see xref:../storage/container_storage_interface/persistent-storage-csi-vsphere.adoc#persistent-storage-csi-vsphere-increase-max-vols-per-node-overview_persistent-storage-csi-vsphere[Increasing maximum volumes per node for vSphere].

[id="ocp-release-notes-storage-vsphere-migrating-cns-vols-between-datastores_{context}"]
==== Migrating CNS volumes between datastores for vSphere is generally available
If you are running out of space in your current datastore, or want to move to a more performant datastore, you can migrate VMware vSphere Cloud Native Storage (CNS) volumes between datastores. This applies to both attached and detached volumes.

This feature was introduced as a Development Preview in {product-title} 4.17, but is now generally available in 4.19.

> Migrating CNS volumes between datastores requires VMware vSphere 8.0.2 or later or vSphere 7.0 Update 3o or later.

Author: Have updated this section as per your suggestions.

|====

> |Feature |4.17 |4.18 |4.19

Reviewer 2: I think we want to add "Increasing the maximum number of nodes per volume for vSphere" as TP in the table.

Author: Done.

|Disabling/enabling storage on vSphere
|Technology Preview
|Technology Preview
|General Availability

> |Increasing max number of nodes per volume for vSphere

Author: Good catch! Fixed.
